src/app/(proper_react)/(redesign)/(public)/layout.tsx

```diff
@@ -7,29 +7,50 @@ import { PublicShell } from "./PublicShell";
 import {
   getAcceptLangHeaderInServerComponents,
   getL10n,
+  getL10nBundles,
 } from "../../../functions/l10n/serverComponents";
 import { headers } from "next/headers";
 import { getCountryCode } from "../../../functions/server/getCountryCode";
 import { getLocale } from "../../../functions/universal/getLocale";
 import { getExperiments } from "../../../functions/server/getExperiments";
 import { getExperimentationId } from "../../../functions/server/getExperimentationId";
 import { getEnabledFeatureFlags } from "../../../../db/tables/featureFlags";
+import { getServerSession } from "../../../functions/server/getServerSession";
+import * as Sentry from "@sentry/nextjs";
 
 export default async function Layout(props: { children: ReactNode }) {
-  const headersList = await headers();
-  const countryCode = getCountryCode(headersList);
   const enabledFeatureFlags = await getEnabledFeatureFlags({
     isSignedOut: true,
   });
   const l10n = getL10n(await getAcceptLangHeaderInServerComponents());
-  const experimentationId = await getExperimentationId(null);
+  const currentLocale = getLocale(
+    getL10nBundles(await getAcceptLangHeaderInServerComponents()),
+  );
+  const session = await getServerSession();
+  const headersList = await headers();
+  const countryCode = getCountryCode(headersList);
+
+  // Check for Nimbus preview mode. Note that this requires a full page reload
+  // to activate: https://nextjs.org/docs/app/api-reference/file-conventions/layout#caveats
+  const nimbusPreviewMode = (await headers()).get("x-nimbus-preview-mode");
+  const experimentationId = await getExperimentationId(session?.user ?? null);
   const experimentData = await getExperiments({
-    experimentationId,
-    countryCode,
-    locale: getLocale(l10n),
-    previewMode: false,
+    experimentationId: experimentationId,
+    countryCode: countryCode,
+    locale: currentLocale,
+    previewMode: nimbusPreviewMode === "true",
```

Comment:
Hmm, I might be missing some context here, but it's not clear to me why callers would need to determine the preview mode in different ways, but I see some instances reading the header added by the middleware, and some directly reading request parameters. Can't `getExperiments` just do that work for us directly? Or are we also using it in emails?

Reply:
In this case, it is because we can’t read request parameters directly from layouts.

Reply:
Right, I suppose that's how we got here. But now that we have that middleware adding the header, I suppose we can just use that everywhere, and incorporate it directly into the function? Would've prevented this bug :)

Reply:
That is true. I’ve created [MNTOR-4011](https://mozilla-hub.atlassian.net/browse/MNTOR-4011) to unify setting the preview mode and ensure we don’t interfere with upcoming experiments.